src/command_modules/azure-cli-batch/azure/cli/command_modules/batch/commands.py

```diff
@@ -29,23 +29,27 @@ def load_command_table(self, _):
     mgmt_path = 'azure.mgmt.batch.operations.{}_operations#{}.'
 
     def get_data_type(name):
-        factory = getattr(factories, "{}_client_factory".format(name))
         return CliCommandType(
             operations_tmpl=data_path.format(name, _operation(name)) + "{}",
-            client_factory=factory,
+            client_factory=get_data_factory(name),
             exception_handler=batch_exception_handler
         )
 
     def get_mgmt_type(name):
-        factory = getattr(factories, "mgmt_{}_client_factory".format(name))
         return CliCommandType(
             operations_tmpl=mgmt_path.format(name, _operation(name)) + "{}",
-            client_factory=factory,
+            client_factory=get_mgmt_factory(name),
             exception_handler=batch_exception_handler
         )
 
+    def get_mgmt_factory(name):
+        return getattr(factories, "mgmt_{}_client_factory".format(name))
+
+    def get_data_factory(name):
+        return getattr(factories, "{}_client_factory".format(name))
+
     # Mgmt Account Operations
-    with self.command_group('batch account', get_mgmt_type('batch_account')) as g:
+    with self.command_group('batch account', get_mgmt_type('batch_account'), client_factory=get_mgmt_factory('batch_account')) as g:
```

Comment:
Was I missing all these client factory references in the cmd groups?
....how was it even running before...? D:

Reply:
Custom_commands aren't supposed to take any kwargs from the regular `command_type`, instead they take in the kwargs from `custom_command_type`, usually defined in `__init__`. 
We had a bug where the client_factory for custom commands were being taken from the regular `command_type`, in which the client factory was attached previously.

Adding a client_factory on the command_group level overrides the (custom)_command_type client_factory hence the fix :)